src/agent/src/mount.rs

```diff
@@ -573,6 +581,92 @@ fn mount_storage(logger: &Logger, storage: &Storage) -> Result<()> {
     )
 }
 
+#[instrument]
+pub fn set_ownership(logger: &Logger, storage: &Storage) -> Result<()> {
+    let logger = logger.new(o!("subsystem" => "change_ownership"));
+
+    // If fsGroup is not set, skip performing ownership change
+    if storage.fs_group == 0 {
+        return Ok(());
+    }
+
+    let mut read_only = false;
+    let opts_vec: Vec<String> = storage.options.to_vec();
+    if opts_vec.contains(&String::from("ro")) {
+        read_only = true;
+    }
+
+    let mount_path = Path::new(&storage.mount_point);
+
+    let mut skip_ownership_change = false;
+    if storage.fs_group_change_policy == FSGroupChangePolicy::OnRootMismatch {
+        if mount_path.metadata()?.gid() == storage.fs_group {
+            let mut mask = RW_MASK;
```

Comment:
Nit: Here can use Rust `let x = if (y){ x1} else {x2}`, but since `clippy` doesn't complain it, don't matter.

Reply:
Thanks, fixed.